cdi/src/main/java/org/infinispan/cdi/util/defaultbean/DefaultBeanExtension.java

```diff
@@ -340,11 +340,12 @@ void afterBeanDiscovery(@Observes AfterBeanDiscovery event, BeanManager manager)
             if (bean == null) {
                 bean = defaultProducerFields.get(e.getKey());
             }
-            if (duplicateDetectionMap.containsKey(e.getValue())) {
+            if (bean != null && duplicateDetectionMap.containsKey(e.getValue())) {
```

Comment:
Please correct me if I'm wrong, but it happened when testing 2 uberjars together (which in turn cased this issue: https://issues.jboss.org/browse/ISPN-5092)?

If I'm correct - we won't be able to support such setup. Here is a relevant part from CDI spec [1]: `If a bean class is deployed in two different bean archives, non-portable behavior results. Portable applications must deploy each bean class in no more than one bean archive.`

[1] http://docs.jboss.org/cdi/spec/1.0/html_single/#beanarchive